pkg/controller/replicaset/replica_set.go

```diff
@@ -591,31 +579,22 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 		return nil
 	}
 
+	// list all pods to include the pods that don't match the rs`s selector
```

Comment:
There is already an open PR for this one: https://github.com/kubernetes/kubernetes/pull/41282